TypeCobol/Compiler/Scanner/Token.cs

```diff
@@ -74,8 +74,14 @@ internal Token(TokenType tokenType, int startIndex, int stopIndex, bool usesVirt
             HasClosingDelimiter = false;
 
             UsesVirtualSpaceAtEndOfLine = usesVirtualSpaceAtEndOfLine;
-        }  
-      
+
+            //Scan Dependent Tokens Inside DataDivision must have their scan state. see #428
+            if (tokensLine is TokensLine tl && tl.ScanState != null && (tokenType == TokenType.PartialCobolWord || 
```

Comment:
ScanState could be exposed in ITokensLine to avoid this cast.

Reply:
ok done in 
64eb1a1